models/audit-log.go

```diff
@@ -0,0 +1,90 @@
+package models
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/wopta/goworkspace/lib"
+)
+
+const (
+	auditTable = "audit"
+	dataset    = "wopta"
+)
+
+// The AuditLog is a data structure that stores request's data.
+// In particular, the Payload should contain data from the request's body.
+//
+// Example:
+//
+//	a := ParseHttpRequest(r)
+//	a.SaveToBigQuery()
+type AuditLog struct {
+	Payload  string    `bigquery:"payload"`
+	Date     time.Time `bigquery:"date"`
+	UserUid  string    `bigquery:"userUid"`
+	Method   string    `bigquery:"method"`
+	Endpoint string    `bigquery:"endpoint"`
+	Role     string    `bigquery:"role"`
+}
+
+func ParseHttpRequest(r *http.Request) (AuditLog, error) {
+	if r.Body != nil {
+		return AuditLog{}, fmt.Errorf("no payload found: body is empty!")
+	}
+
+	defer r.Body.Close()
+
+	body, err := io.ReadAll(r.Body)
+	if err != nil {
+		return AuditLog{}, fmt.Errorf("cannot retrieve the payload: %v", err)
+	}
+
+	userUid := r.Header.Get("userUid")
```

Comment:
The userUid wont't be available from the Header. We have to extract it from the authorization token. Check `lib.VerifyUserIdToken`